spring-data-relational/src/main/java/org/springframework/data/relational/core/conversion/WritingContext.java

```diff
@@ -63,7 +63,9 @@
 		this.aggregateChange = aggregateChange;
 		this.rootIdValueSource = IdValueSource.forInstance(root,
 				context.getRequiredPersistentEntity(aggregateChange.getEntityType()));
-		this.paths = context.findPersistentPropertyPaths(entityType, (p) -> p.isEntity() && !p.isEmbedded());
+		this.paths = context.findPersistentPropertyPaths(entityType, (p) -> p.isEntity() && !p.isEmbedded()) //
+				.filter(PersistentPropertyPathExtension::isWritable) //
+				.stream().toList();
```

Comment:
Why do we need a stream then a list here? Is this the most efficient API to use to produce a `List`?